bee-network/src/identity.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A module that deals with network identities.
+
+use crate::consts::ID_LENGTH;
+
+use crypto::{hashes::sha, signatures::ed25519};
+
+use std::{
+    fmt,
+    sync::{Arc, RwLock},
+};
+
+/// A type that represents a local identity, which is able to sign messages.
+#[derive(Clone)]
+pub struct LocalIdentity {
+    secret_key: Arc<RwLock<ed25519::SecretKey>>,
+    identity: Identity,
+}
+
+impl LocalIdentity {
+    /// Creates a new local identity.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Creates a local identity from a 'base58' encoded ED25519 private key.
+    pub fn from_bs58_secret_key_str(sk: &str) -> Self {
+        let sk = bs58::decode(sk).into_vec().expect("error restoring secret key");
```

Comment:
The names `sk` and `secret_key` are a little confusing here.

Reply:
Do you want me to use `private_key` instead? (I think the only reason why the term `secret_key` is also used very commonly is because it shortens to `sk` which allows to unambiguously assign `pk` to "public key").

Reply:
I think I would probably use `private_key` -- it's also what's in the the  doc-comment, so I think it makes the most sense.